contractcourt/chain_arbitrator.go

```diff
@@ -505,6 +490,30 @@ func (c *ChainArbitrator) Stop() error {
 	return nil
 }
 
+// notifyChanClosed is a helper function that cleans up the internal state for a
+// cooperatively closed channel.
+func (c *ChainArbitrator) notifyChanClosed(chanPoint wire.OutPoint,
+	remotePub *btcec.PublicKey) error {
+
+	c.Lock()
+	delete(c.activeChannels, chanPoint)
+
+	chainWatcher, ok := c.activeWatchers[chanPoint]
+	if ok {
+		// Since the chainWatcher is calling notifyChanClosed, we must
+		// stop it in a goroutine to not deadlock.
+		go chainWatcher.Stop()
+	}
+	delete(c.activeWatchers, chanPoint)
+	c.Unlock()
+
+	// Since MarkChanFullyClosed is not necessary when closing a channel
```

Comment:
If we instead just call `MarkChanFullyClosed` also on cooperative close, then we can avoid exporting `PruneLinkNodes`. It also makes the code less confusing if `MarkChanFullyClosed` is called for all channel close types.

Reply:
Fixed.